dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/converter/RelationshipTrackerConverterService.java

```diff
@@ -179,8 +180,10 @@ else if ( relationshipType.getToConstraint().getRelationshipEntity().equals( PRO
 
         toRelationship.setFrom( fromItem );
         toRelationship.setTo( toItem );
-        toRelationship.setKey( RelationshipUtils.generateRelationshipKey( toRelationship ) );
-        toRelationship.setInvertedKey( RelationshipUtils.generateRelationshipInvertedKey( toRelationship ) );
+        RelationshipKey relationshipKey = RelationshipKeySupport.getRelationshipKey( fromRelationship,
```

Comment:
why do we now use `fromRelationship` and before we were using `toRelationship` to get the RelationshipKey?

Reply:
I think it was conceptually wrong before even though it was working.
We are converting `fromRelationship` into `toRelationship` so we should use `fromRelationship` to generate all the fields to create the target object.

Using `toRelationship` is working only if we generate the key after having populated all the other fields, `fromRelationship` already have all those fields populated